buildroot-external/package/generic_raw_uart/0003-Linux-6.12-gpio-data.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/kernel/hb_rf_eth.c b/kernel/hb_rf_eth.c
+index 9d949c003..c8d5f15bf 100644
+--- a/kernel/hb_rf_eth.c
++++ b/kernel/hb_rf_eth.c
+@@ -715,7 +715,7 @@ static int __init hb_rf_eth_init(void)
+   gc.base = -1;
+   gc.can_sleep = false;
+ 
+-  err = gpiochip_add(&gc);
++  err = gpiochip_add_data(&gc, NULL);
```

Comment:
And what about a similar `#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 0, 0)` preprocessor statement here to make sure the sources still compile for other kernel versions. And also please check at which kernel version `gpiochip_add_data()` was added and then restrict the `#if LINUX_VERSION_CODE` version check to that specific kernel version.